src/commands.cc

```diff
@@ -588,6 +588,9 @@ static void ensure_all_buffers_are_saved()
     if (it == end)
         return;
 
+    if (not context.buffer().is_modified())
+        context.change_buffer(**it);
```

Comment:
this is a good change, I've been running it for a while.

However we should add `context.push_jump()` here to allow the user to go back with `<c-o>` or `ga`